openlibrary/tests/solr/test_query_utils.py

```diff
@@ -85,3 +86,14 @@ def fn(query: str) -> str:
     assert fn('no fields here!') == 'no fields here!'
     # This is non-ideal
     assert fn('NOT title:foo bar') == 'NOT title:foo bar'
+
+
+def test_luqum_replace_fields():
+    def replace_work_prefix(string: str):
+        return string.partition(".")[2] if string.startswith("work.") else string
+
+    def fn(query: str) -> str:
+        return luqum_replace_field(luqum_parser(query), replace_work_prefix)
+
+    assert fn('work.title:Bob') == 'title:Bob'
+    assert fn('title:Joe') == 'title:Joe'
```

Comment:
Can we use some complex queries here? These tests would also pass this function ;)

```py
def luqum_replace_field(query, replacer):
    return replacer(query)
```